test/integ/cli/test_cli_incremental_download.py

```diff
@@ -0,0 +1,596 @@
+                time.sleep(5)
+
+        print(
+            f"[dep_data_flow] Job completed after {incremental_download_iteration_number} download iterations"
+        )
+
+        # Run final incremental download to ensure all files are captured
+        time.sleep(5)
+        final_result = incremental_download_test.run_incremental_download_without_storage_profiles(
+            unique_data_dir, test_name="dep_data_flow"
+        )  # Reduced from 1200
+        assert final_result.returncode == 0, (
+            f"Final incremental download failed: {final_result.stderr}"
+        )
+
+        # Verify expected output files from dep_data_flow template
+        # Based on template: Step1 + Step1-2 (frames 8-11) + Step1-2-3 + Step1-2-4 + Step1-2-34-5 = 8 files total
+        expected_files = {
+            "Step1.out": "2. Processed in Step1",
+            "Step1-2.8.out": "3.8 Processed in Step1-2.8",
+            "Step1-2.9.out": "3.9 Processed in Step1-2.9",
+            "Step1-2.10.out": "3.10 Processed in Step1-2.10",
+            "Step1-2.11.out": "3.11 Processed in Step1-2.11",
+            "Step1-2-3.out": "4. Processed in Step1-2-3",
+            "Step1-2-4.out": "4. Processed in Step1-2-4",
+            "Step1-2-34-5.out": "5. Processed in Step1-2-34-5",
+        }
+
+        # Check unique data dir explicitly (these might be output files)
+        data_files = list(unique_data_path.glob("**/*.out"))
+
+        # Remove duplicates by file path (not just name) to avoid counting the same file twice
+        unique_file_paths = list(set(data_files))
+        downloaded_files = list(unique_file_paths)
+
+        verified_files = []
+
+        for downloaded_file in downloaded_files:
+            filename = downloaded_file.name
+            if filename in expected_files:
+                expected_marker = expected_files[filename]
+                content = downloaded_file.read_text()
+                # Verify file contains the original input
+                if "1. Input to CreateJob from data_dir" in content and expected_marker in content:
+                    verified_files.append(filename)
+                else:
+                    print(f"[dep_data_flow] WARNING: {filename} missing expected content")
+
+        # Verify all 8 expected files were created and have correct content
+        assert len(verified_files) == 8, (
+            f"Expected exactly 8 verified output files, found {len(verified_files)}: {verified_files}"
+        )
+
+        # Verify we have the complete set of expected files
+        missing_files = set(expected_files.keys()) - set(verified_files)
+        assert not missing_files, f"Missing expected files: {missing_files}"
+
+        print(
+            f"[dep_data_flow] Successfully verified all {len(verified_files)} expected output files with correct content"
+        )
+
+    finally:
+        # Clean up test-specific output files after validation (always runs, even if test fails)
```

Comment:
If you use the pytest tmp_path fixture (https://docs.pytest.org/en/6.2.x/tmpdir.html) and make all directories as subdirectories of that, you can remove all custom cleanup code because pytest will do that for you.